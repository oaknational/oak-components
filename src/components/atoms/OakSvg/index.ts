export { OakSvg } from "./OakSvg";
export type { OakSvgProps, OakSvgNames } from "./OakSvg";
