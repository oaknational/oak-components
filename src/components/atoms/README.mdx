# Atoms

## Rules for components in this folder

- Can import other atoms
- Imports from molecules and organisms are forbidden!
- Generic
- Unstyled
- Avoid embedded logic and state
- No sub-components
  (eg. Box)
