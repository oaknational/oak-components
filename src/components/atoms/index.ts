export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
export * from "./OakIcon";
export * from "./OakGrid";
export * from "./OakGridArea";
export * from "./OakThemeProvider";
export * from "./OakMaxWidth";
export * from "./OakCloudinaryImage";
export * from "./OakKbd";
export * from "./OakGlobalStyle";
export * from "./OakAnchorTarget";
export * from "./OakFieldSet";
