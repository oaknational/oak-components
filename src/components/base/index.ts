export * from "./OakBox";
export * from "./OakFlex";
export * from "./OakBaseCard";
export * from "./OakSpan";
export * from "./OakForm";
export * from "./OakImage";
export * from "./OakTypography";
export * from "./OakHeading";
export * from "./OakP";
export * from "./OakUL";
export * from "./OakOL";
export * from "./OakLI";
export * from "./OakLabel";
