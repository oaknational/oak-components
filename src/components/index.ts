export * from "./base";
// export * from "./ui"; // uncomment this line once there are components in this folder
// export * from "./integrated"; // uncomment this line once there are components in this folder
