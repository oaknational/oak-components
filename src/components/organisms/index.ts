export * from "./pupil";
export * from "./teacher";
export * from "./OakSubjectIcon";
export * from "./OakPrimaryNavItem";
export * from "./OakPrimaryNav";
export * from "./OakCookieBanner";
export * from "./OakCookieSettingsModal";
export * from "./OakCookieConsentProvider";
export * from "./OakCookieConsent";
