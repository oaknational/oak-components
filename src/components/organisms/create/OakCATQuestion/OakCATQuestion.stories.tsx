import React from "react";
import { StoryObj, Meta } from "@storybook/react";

import { OakCATQuestion } from "./OakCATQuestion";

import { OakBox, OakP, OakUL } from "@/components/atoms";
import { OakMultilineText } from "@/components/molecules";

const meta: Meta<typeof OakCATQuestion> = {
  component: OakCATQuestion,
  tags: ["autodocs"],
  argTypes: {
    questionNumber: { control: "number" },
    status: {
      control: { type: "select" },
      options: ["error", "selected", "neutral"],
    },
    initialOpen: { control: "boolean" },
  },
  parameters: {
    controls: {
      include: ["questionNumber", "status", "initialOpen"],
    },
  },
  decorators: [
    (Story) => (
      <OakBox $background={"grey20"} $pa="inner-padding-xl" $width={"100%"}>
        <OakUL>
          <Story />
        </OakUL>
      </OakBox>
    ),
  ],
};

export default meta;

type Story = StoryObj<typeof OakCATQuestion>;

export const Default: Story = {
  render: (args) => <OakCATQuestion {...args} />,
  args: {
    questionNumber: 1,
    status: "neutral",
    questionTypeInput: (
      <OakBox $background={"aqua50"} $pa="inner-padding-ssx">
        Dummy Question Type Input
      </OakBox>
    ),
    questionTextInput: (
      <OakBox $background={"aqua50"} $pa="inner-padding-s">
        Dummy Question Text Input
      </OakBox>
    ),
    hintInput: (
      <OakBox $background={"aqua50"}>
        <OakBox
          $background={"amber50"}
          $pa="inner-padding-ssx"
          $height={"all-spacing-12"}
        >
          Dummy Hint Input
        </OakBox>
        <OakP>Character count 0/200</OakP>
      </OakBox>
    ),
    feedbackInput: (
      <OakBox $background={"aqua50"}>
        <OakBox
          $background={"amber50"}
          $pa="inner-padding-ssx"
          $height={"all-spacing-12"}
        >
          Dummy Feedback Input
        </OakBox>
        <OakP>Character count 0/200</OakP>
      </OakBox>
    ),
    answersSection: (
      <OakBox
        $background={"mint50"}
        $pa="inner-padding-s"
        $height={"all-spacing-20"}
      >
        Dummy answers Section
      </OakBox>
    ),
  },
};

export const InitiallyOpen: Story = {
  render: (args) => <OakCATQuestion {...args} />,
  args: {
    initialOpen: true,
    questionNumber: 1,
    status: "neutral",
    questionTypeInput: (
      <OakBox $background={"aqua50"} $pa="inner-padding-ssx">
        Dummy Question Type Input
      </OakBox>
    ),
    questionTextInput: (
      <OakBox $background={"aqua50"} $pa="inner-padding-s">
        Dummy Question Text Input
      </OakBox>
    ),
    hintInput: (
      <OakBox $background={"aqua50"}>
        <OakBox
          $background={"amber50"}
          $pa="inner-padding-ssx"
          $height={"all-spacing-12"}
          $width={"all-spacing-20"}
        >
          Dummy Hint Input
        </OakBox>
        <OakP>Character count 0/200</OakP>
      </OakBox>
    ),
    feedbackInput: (
      <OakBox $background={"aqua50"}>
        <OakBox
          $background={"amber50"}
          $pa="inner-padding-ssx"
          $height={"all-spacing-12"}
          $width={"all-spacing-20"}
        >
          Dummy Feedback Input
        </OakBox>
        <OakP>Character count 0/200</OakP>
      </OakBox>
    ),
    answersSection: (
      <OakBox
        $background={"mint50"}
        $pa="inner-padding-s"
        $height={"all-spacing-20"}
      >
        Dummy answers Section
      </OakBox>
    ),
  },
};

export const WithMultilineText: Story = {
  render: (args) => <OakCATQuestion {...args} />,
  args: {
    questionNumber: 1,
    status: "neutral",
    questionTypeInput: (
      <OakBox $background={"aqua50"} $pa="inner-padding-ssx">
        Dummy Question Type Input
      </OakBox>
    ),
    questionTextInput: (
      <OakBox $background={"aqua50"} $pa="inner-padding-s">
        Dummy Question Text Input
      </OakBox>
    ),
    hintInput: (
      <OakMultilineText
        charLimit={100}
        name="hintInput"
        id="hintInput"
        $height={"all-spacing-24"}
      />
    ),
    feedbackInput: (
      <OakMultilineText
        charLimit={100}
        name="feedbackInput"
        id="feedbackInput"
        $height={"all-spacing-24"}
      />
    ),
    answersSection: (
      <OakBox
        $background={"mint50"}
        $pa="inner-padding-s"
        $height={"all-spacing-20"}
      >
        Dummy answers Section
      </OakBox>
    ),
  },
};
