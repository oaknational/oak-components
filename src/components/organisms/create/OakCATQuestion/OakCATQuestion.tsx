import React, { ReactNode } from "react";
import styled from "styled-components";

import { OakFlex, OakFlexProps } from "@/components/atoms/OakFlex";
import { parseColor } from "@/styles/helpers/parseColor";
import { OakCombinedColorToken } from "@/styles";
import { OakLabel, OakSpan } from "@/components/atoms";
import { InternalUnstyledChevronAccordion } from "@/components/molecules/InternalUnstyledChevronAccordion";
import { parseBorderRadius } from "@/styles/helpers/parseBorderRadius";

type StyledOakFlexProps = {
  $statusColor: OakCombinedColorToken;
} & OakFlexProps;

const StyledOakFlex = styled(OakFlex)<StyledOakFlexProps>`
  border-left: 8px solid ${(props) => parseColor(props.$statusColor)};
  border-top-right-radius: ${parseBorderRadius("border-radius-m2")};
  border-bottom-right-radius: ${parseBorderRadius("border-radius-m2")};
`;

const UnstyledLI = styled.li`
  list-style: none;
`;

type Status = "error" | "selected" | "neutral";

export type OakCATQuestionProps = {
  questionNumber: number;
  status: Status;
  initialOpen?: boolean;
  questionTypeInput: ReactNode;
  questionTextInput: ReactNode;
  hintInput: ReactNode;
  feedbackInput: ReactNode;
  answersSection: ReactNode;
};

export const OakCATQuestion = (props: OakCATQuestionProps) => {
  const {
    questionNumber,
    questionTypeInput,
    questionTextInput,
    hintInput,
    feedbackInput,
    answersSection,
    initialOpen = false,
    status = "neutral",
  } = props;

  const statusColorMap: Record<Status, OakCombinedColorToken> = {
    error: "red",
    selected: "mint110",
    neutral: "grey40",
  };

  const header = (
    <OakFlex $gap={"spacing-12"} $alignItems={"center"} $width={"100%"}>
      {`${questionNumber}.`}
      {questionTypeInput}
    </OakFlex>
  );

  const body = (
    <OakFlex $flexDirection={"column"} $width={"100%"} $gap={"spacing-48"}>
      {/* this has the effect of stretching the container and the items within */}
      <OakFlex $alignSelf={"stretch"} $flexDirection={"column"}>
        {answersSection}
      </OakFlex>

      <OakFlex
        $ph="spacing-32"
        $pv="spacing-24"
        $background={"grey10"}
        $borderRadius={"border-radius-m2"}
        $borderColor={"bg-neutral"}
        $ba={"border-solid-s"}
        $alignSelf={"flex-start"}
      >
        <OakFlex $gap="spacing-20">
          <OakLabel>
            <OakSpan $color={"text-subdued"} $font={"heading-light-7"}>
              Hint
            </OakSpan>
            {hintInput}
          </OakLabel>

          <OakLabel>
            <OakSpan $color={"text-subdued"} $font={"heading-light-7"}>
              Feedback
            </OakSpan>
            {feedbackInput}
          </OakLabel>
        </OakFlex>
      </OakFlex>
    </OakFlex>
  );

  return (
    <UnstyledLI>
      <StyledOakFlex
        $width={"100%"}
        $justifyContent={"flex-start"}
        $background={"bg-primary"}
        $pa={"spacing-32"}
        $statusColor={statusColorMap[status]}
        $flexDirection={"column"}
      >
        <InternalUnstyledChevronAccordion
          id={`question-${questionNumber}`}
          header={header}
          subheader={questionTextInput}
          content={body}
          $flexDirection={"column"}
          $justifyContent={"flex-start"}
          initialOpen={initialOpen}
          ariaLabelClose={`expand question ${questionNumber}`}
          ariaLabelOpen={`collapse question ${questionNumber}`}
        />
      </StyledOakFlex>
    </UnstyledLI>
  );
};
