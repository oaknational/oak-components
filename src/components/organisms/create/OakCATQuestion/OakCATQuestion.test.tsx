import React from "react";
import "@testing-library/jest-dom";

import { OakCATQuestion } from "./OakCATQuestion";

import renderWithTheme from "@/test-helpers/renderWithTheme";
import { OakThemeProvider } from "@/components/atoms";
import { oakDefaultTheme } from "@/styles";

describe("OakCATQuestion", () => {
  it("renders", () => {
    const { getByText } = renderWithTheme(
      <OakCATQuestion
        data-testid="test"
        questionNumber={1}
        status="neutral"
        questionTypeInput={<div>Question Type Input</div>}
        questionTextInput={<div>Question Text Input</div>}
        hintInput={<div>Hint Input</div>}
        feedbackInput={<div>Feedback Input</div>}
        answersSection={<div>Answers Section</div>}
        hintInputId="hintInputId"
        feedbackInputId="feedbackInputId"
      />,
    );
    expect(getByText("1.")).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = renderWithTheme(
      <OakThemeProvider theme={oakDefaultTheme}>
        <OakCATQuestion
          data-testid="test"
          questionNumber={1}
          status="neutral"
          questionTypeInput={<div>Question Type Input</div>}
          questionTextInput={<div>Question Text Input</div>}
          hintInput={<div>Hint Input</div>}
          feedbackInput={<div>Feedback Input</div>}
          answersSection={<div>Answers Section</div>}
          hintInputId="hintInputId"
          feedbackInputId="feedbackInputId"
        />
      </OakThemeProvider>,
    );
    expect(container).toMatchSnapshot();
  });
});
