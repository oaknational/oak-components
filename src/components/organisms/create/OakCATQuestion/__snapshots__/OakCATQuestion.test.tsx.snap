// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OakCATQuestion matches snapshot 1`] = `
.c1 {
  width: 100%;
  padding: 2rem;
  background: #ffffff;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c4 {
  padding-top: 0.75rem;
  padding-bottom: 0.75rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c5 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c7 {
  width: 100%;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c9 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c9:hover {
  cursor: pointer;
}

.c13 {
  margin-right: 0.75rem;
  border-radius: 0.25rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c14 {
  position: relative;
  width: 2rem;
  min-width: 2rem;
  height: 2rem;
  min-height: 2rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c16 {
  position: relative;
  overflow: auto;
  font-family: --var(google-font),Lexend,sans-serif;
  overflow: auto;
}

.c19 {
  padding-left: 2rem;
  padding-right: 2rem;
  padding-top: 1.5rem;
  padding-bottom: 1.5rem;
  background: #f9f9f9;
  border: 0.063rem solid;
  border-color: #f2f2f2;
  border-radius: 0.5rem;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.c8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.75rem;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 3rem;
}

.c18 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: stretch;
  -ms-flex-item-align: stretch;
  align-self: stretch;
}

.c20 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: start;
  align-self: flex-start;
}

.c21 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.25rem;
}

.c15 {
  object-fit: contain;
}

.c23 {
  color: #575757;
  font-family: --var(google-font),Lexend,sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.25rem;
  -webkit-letter-spacing: 0.0115rem;
  -moz-letter-spacing: 0.0115rem;
  -ms-letter-spacing: 0.0115rem;
  letter-spacing: 0.0115rem;
}

.c22 {
  font-family: --var(google-font),Lexend,sans-serif;
}

.c11 {
  font: inherit;
  color: inherit;
  border: none;
  background: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  outline: none;
  padding: 0;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-family: --var(google-font),Lexend,sans-serif;
}

.c12:hover .chevron-icon {
  -webkit-filter: invert(34%) sepia(0%) saturate(698%) hue-rotate(158deg) brightness(95%) contrast(89%);
  filter: invert(34%) sepia(0%) saturate(698%) hue-rotate(158deg) brightness(95%) contrast(89%);
}

.c12:focus-visible .focus-outline {
  box-shadow: 0 0 0 0.125rem rgba(255,229,85,100%), 0 0 0 0.3rem rgba(87,87,87,100%);
}

.c3 {
  border-left: 8px solid #cacaca;
  border-top-right-radius: 0.5rem;
  border-bottom-right-radius: 0.5rem;
}

.c0 {
  list-style: none;
}

<li
  className="c0"
>
  <div
    className="c1 c2 c3"
  >
    <div
      className="c4 c2"
    >
      <div
        className="c5 c6"
      >
        <div
          className="c7 c8"
        >
          1.
          <div>
            Question Type Input
          </div>
        </div>
        <div
          className="c5"
        >
          <button
            aria-expanded={false}
            aria-label="collapse question 1"
            className="c9 c10 c11  c12"
            id="question-1"
            onClick={[Function]}
            type="button"
          >
            <div
              className="c13 focus-outline"
            >
              <div
                className="c14 chevron-icon"
                style={
                  {
                    "transform": "none",
                    "transition": "all 0.3s ease 0s",
                  }
                }
              >
                <img
                  alt="An arrow to indicate whether the item is open or closed"
                  className="c15"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  onError={[Function]}
                  onLoad={[Function]}
                  src="https://res.cloudinary.com/mock-cloudinary-cloud/image/upload/v1699953557/icons/botfld6brychmttwtv6u.svg"
                  style={
                    {
                      "bottom": 0,
                      "color": "transparent",
                      "height": "100%",
                      "left": 0,
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                      "width": "100%",
                    }
                  }
                />
              </div>
            </div>
          </button>
        </div>
      </div>
      <div>
        Question Text Input
      </div>
      <div
        className="c16"
      >
        <div
          aria-labelledby="question-1"
          className="c5"
          hidden={true}
          role="region"
        >
          <div
            className="c7 c17"
          >
            <div
              className="c5 c18"
            >
              <div>
                Answers Section
              </div>
            </div>
            <div
              className="c19 c20"
            >
              <div
                className="c5 c21"
              >
                <label
                  className="c22"
                >
                  <span
                    className="c23"
                  >
                    Hint
                  </span>
                  <div>
                    Hint Input
                  </div>
                </label>
                <label
                  className="c22"
                >
                  <span
                    className="c23"
                  >
                    Feedback
                  </span>
                  <div>
                    Feedback Input
                  </div>
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</li>
`;
